packages/multicast_dns/lib/src/resource_record.dart

```diff
@@ -0,0 +1,393 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:meta/meta.dart';
+import 'package:multicast_dns/src/constants.dart';
+import 'package:multicast_dns/src/packet.dart';
+
+// TODO(dnfield): Probably should go with a real hashing function here
+// when https://github.com/dart-lang/sdk/issues/11617 is figured out.
+const int _seedHashPrime = 2166136261;
+const int _multipleHashPrime = 16777619;
+
+int _combineHash(int current, int hash) =>
+    (current & _multipleHashPrime) ^ hash;
+
+int _hashValues(List<int> values) {
+  assert(values != null);
+  assert(values.isNotEmpty);
+
+  return values.fold(
+    _seedHashPrime,
+    (int current, int next) => _combineHash(current, next),
+  );
+}
+
+/// Enumeration of support resource record types.
+class ResourceRecordType {
+  // This class is intended to be used as a namespace, and should not be
+  // extended directly.
+  factory ResourceRecordType._() => null;
+
+  /// An IPv4 Address record (1).
```

Comment:
```suggestion
  /// An IPv4 Address record. It has a value of 1.
```
Because otherwise the (1) looks like a footnote or something. Same for the other constants in this class.

Also, a single character isn't really good enough as a variable name.  Maybe "ipv4Address" or "addressIpv4"?

Reply:
On these I wanted to keep it aligned to the RFC's way of referring to them for clarity.  I suppose I could just add that in the doc comment instead though.